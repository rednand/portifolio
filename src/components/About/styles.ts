'use client'

import styled from 'styled-components'

export const Wrapper = styled.section`
  background-color: #2c2a32;
  height: 100vh;
  padding: 10vh;
  font-family: 'Segoe UI', Roboto, 'Helvetica Neue', sans-serif;
  margin: auto;
  scroll-snap-align: start;
`

export const Description = styled.h3`
  font-size: 4rem;
  line-height: 120%;
  margin-top: 8%;
  width: 70%;
  color: #fefefe;
  font-weight: 400;
`

export const Text = styled.p`
  font-size: 1.7rem;
  line-height: 150%;
  width: 60%;
  padding-bottom: 50px;
  padding-top: 2%;
  color: #fefefe;
  font-weight: 100;
  border-bottom: 10px solid #48d8a4;
`
