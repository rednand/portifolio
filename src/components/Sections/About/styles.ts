'use client'

import styled from 'styled-components'

export const Wrapper = styled.section`
  background-color: #2c2a32;
  height: 100vh;
  padding: 10vh;
  font-family: 'Segoe UI', Roboto, 'Helvetica Neue', sans-serif;
  margin: auto;
  scroll-snap-align: start;

  @media (max-width: 600px) {
    padding: 5vh;
  }
`

export const Description = styled.h3`
  line-height: 120%;
  margin-top: 7%;
  width: 70%;
  color: #fefefe;
  font-size: 2.5rem;
  font-family: 'Segoe UI', Roboto, 'Helvetica Neue', sans-serif;
  font-weight: 700;
  margin-bottom: 3%;

  @media (max-width: 600px) {
    font-size: 2rem;
    margin: 8% 0;
  }
`

export const Text = styled.p`
  font-size: 1.7rem;
  line-height: 150%;
  width: 60%;
  padding-bottom: 50px;
  color: #fefefe;
  font-weight: 100;
  border-bottom: 10px solid #48d8a4;

  @media (max-width: 600px) {
    font-size: 1rem;
    width: 100%;
  }
`
