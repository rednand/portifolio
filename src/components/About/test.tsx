import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import About from './about'

describe('About Component', () => {
  it('renders the About component correctly', () => {
    const { container } = render(<About />)

    expect(screen.getByText('Sobre')).toBeInTheDocument()

    expect(
      screen.getByText(/Back in 2012, I decided to try my hand/i)
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<About />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#2c2a32' })
  })
})
