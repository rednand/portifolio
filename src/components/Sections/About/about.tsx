import * as Style from './styles'

export default function About() {
  return (
    <Style.Wrapper>
      <section>
        <div>
          <Style.Description>Sobre</Style.Description>
          <Style.Text>
            Minha jornada na tecnologia começou no início de 2019, durante um
            mestrado em Bioinformática, onde descobri o gosto pela programação
            com Python e Perl. Em 2020, decidi focar no desenvolvimento de
            sistemas e iniciei uma nova graduação em Análise e Desenvolvimento
            de Sistemas. Desde então, venho me especializando em tecnologias
            como ReactJS, TypeScript, JavaScript e React Native, além de
            ferramentas modernas como NextJS, Redux, React Query e Material UI.
            Atualmente, atuo na Multiplica Crédito e Investimento, desenvolvendo
            sistemas financeiros com arquitetura de micro frontends.
            Recentemente, finalizei uma pós-graduação em Desenvolvimento Web
            Full Stack pela PUC Minas e sigo em constante evolução, sempre
            buscando aprender e transformar boas ideias em soluções reais e
            acessíveis.
          </Style.Text>
        </div>
      </section>
    </Style.Wrapper>
  )
}
