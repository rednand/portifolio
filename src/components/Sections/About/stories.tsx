import { Meta, StoryObj } from '@storybook/react'
import About from './about'

export default {
  title: 'About',
  component: About,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryObj = {}
