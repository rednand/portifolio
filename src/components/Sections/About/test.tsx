import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import About from './about'

describe('About Component', () => {
  it('renders the About component correctly', () => {
    const { container } = render(<About />)

    expect(screen.getByText('Sobre')).toBeInTheDocument()

    expect(
      screen.getByText(/Minha jornada na tecnologia começou no início de 2019/i)
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<About />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#2c2a32' })
  })
})
