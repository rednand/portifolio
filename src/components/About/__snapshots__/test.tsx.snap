// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component renders the About component correctly 1`] = `
.c0 {
  background-color: #2c2a32;
  height: 100vh;
  padding: 10vh;
  font-family: 'Segoe UI',Roboto,'Helvetica Neue',sans-serif;
  margin: auto;
}

.c1 {
  font-size: 4rem;
  line-height: 120%;
  margin-top: 8%;
  width: 70%;
  color: #fefefe;
  font-weight: 400;
}

.c2 {
  font-size: 1.7rem;
  line-height: 150%;
  width: 60%;
  padding-bottom: 50px;
  padding-top: 2%;
  color: #fefefe;
  font-weight: 100;
  border-bottom: 10px solid #48d8a4;
}

<section
  class="c0"
>
  <section>
    <div>
      <h3
        class="c1"
      >
        Sobre
      </h3>
      <p
        class="c2"
      >
        Back in 2012, I decided to try my hand at creating custom Tumblr themes and tumbled head first into the rabbit hole of coding and web development. Fast-forward to today, and I’ve had the privilege of building software for an advertising agency, a start-up, a student-led design studio, and a huge corporation. My main focus these days is building products and leading projects for our clients at Upstatement. In my free time I've also released an online video course that covers everything you need to know to build a web app with the Spotify API. When I’m not at the computer, I’m usually rock climbing, hanging out with my wife and two cats, or running around Hyrule searching for Korok seeds .
      </p>
    </div>
  </section>
</section>
`;
